### DISCUSSION TITLE
Hourly weather data as a circle

### DISCUSSION THREAD

**Discussion by esetleges** [breezy-weather/breezy-weather#1154] [Ideas]
### Describe your idea

Design:
There was never made better design for weather app than Partly Cloudy app for iOS:
https://weandthecolor.com/partly-cloudy-iphone-weather-app-by-timm-kekeritz/13331?amp=1

https://www.cultofmac.com/182905/partly-cloudy-might-be-the-beautiful-ios-weather-app-youve-been-waiting-for/

I was always wondering noone else put weather data on a circle, instead of linar design which is complicated to oversee. While it is (was) briliant: You could see visually in no time the whole weather situation.

- 1st Outer cirlcle: Daytime/Nightime lenght (light/dark hours)
- 2nd outer circle (plus filling color): Temperatures coded by colors (blue- icy, green- warmer, yellow - warmer than green, red - hot)
- Inner grey lines: wind speed prec.
- Inner blue: rain prec 

- right bottom corner: you could flip view between 12h/24h/or week 
- Anchor: rolling/uncovering the further weather data (time shown in middle circle)

May you could contact the developer to 'licence his design' or just get some ideas. May you find the this ios app and emulate it (it is over 10 year old) to see how it works live (hope still pulls weather data)

PWS (Personal Weather Station)
- Add personal weather station info (widget/app) with API (weatherunderground/ambient weather/etc)

Daily widget:
- change text size
- custom info: alerts by an icon (next to $al$, $als$)

Sources:
- add more weather  sources if possible
- when you change weather source (list view) show all current weather info since more easy to swich to the 'right/correct' one for location

More ideas later.

Thanks for the app!

Cheers!

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> I was always wondering noone else put weather data on a circle, instead of linar design which is complicated to oversee

There are now better design than 10 years ago. Next 24 hours-window-fitting line charts from latest iOS weather app are great and easy to read. It will be implemented later in #13 

As for your summary circle, good luck with your new app! You can easily re-use our data from Gadgetbridge WeatherSpec or wait for ours in #863 so you only focus on the design.

However, be careful as hourly data is not always hour by hour. Sometimes it starts with hour by hour for the first 48 hours, then switch to 3 hour by 3 hour for the next 3 days, and finally to 6 hour by 6 hour. So you might need to adjust the proportions of your circle accordingly.

Feel free to let us know your progress!

**Reply by esetleges** [reply]
Hi there,


Thanks for quick response!

No plan for making own app, it was just a suggestion to you.


I would discuss is there any better weather infograpics to understand forecast as a whole (temperature+daylight+wind+rain - in general the most important data you want to know) than that over 10 year old design. 

I can read in a weather app(s) 'today is 2 degree warmer'. But in that app eveything is visually presented (colors+shapes for eyes). That is much more easy and better to understand (for the brain) than texts, rows of data and/or lines. 

Im not saying that your (or any weather app) is not good in design. Im only said that Party Cloudy idea/design/infograpics are still the very best. 


Cheers!

**Reply by papjul** [reply]
> is there any better weather infograpics to understand forecast as a whole (temperature+daylight+wind+rain - in general the most important data you want to know) than that over 10 year old design

I can't answer that question because we have different opinions on the matter. I don't find this design easier to read.

IMO, this UV index chart is easier to read:
![Screenshot_20240623_113359](https://github.com/breezy-weather/breezy-weather/assets/1449159/a9650ca8-d934-48f9-9966-a7bd79930719)

Than this circle which doesn't mention any UV and lacks many of our other weather data as far as I can tell:
![Screenshot_20240623_113520](https://github.com/breezy-weather/breezy-weather/assets/1449159/fde768cd-44d4-40b4-9d7e-e1db109ecbb8)

You lack any details until you tap on the hour. The small color circle only gives rough information, and in the case just above, the greenish-yellowish color doesn't tell me anything as there is no legend. I guess it goes better when you get used to, but it's far from being as readable as a line chart with axis, color gradients, textual descriptions and line thresholds, which will be the way to go in the future of Breezy Weather.

Since we have different opinions, you have to make your own researches and find which other weather app you like best ;)

Alternatively, you can make your own design. Like I told you, our weather data can be shared with others apps, so if you take the time to design it, you can just focus on the design and you don't have to worry about getting the data from weather sources since we just push it to your app as soon as it's refreshed ;)

**Reply by esetleges** [reply]
Hey again, no plan for own app, and not manditory to implement circle design at all.


<picture>https://weandthecolor.com/wp-content/uploads/2012/08/Partly-Cloudy-iPhone-Weather-App-by-Timm-Kekeritz-63217.jpg</picture>

Greenish-yellowis thing temperature between 13-15 degrees something like. If I rember well, under 0 blue, 3-12 green, 15-24 yellow,  and above 28 red. Anything between is mixed color. (Im not sure with exact color scale, but it does not matter too much again it visualize the temperature)


UV index (im not sure that time YR.no publish that info). However you can add/exhange your own cicle like UV index, cloudiness, etc.


And again with this infographic design I see/understand in no time 4 different weather data.


- I would be more happy to implement into Breezy ability to add PWS. ( no plan for make that app neither : )

**Reply by papjul** [reply]
It's true that you can see more different data at once. However, it's more detailed with a more efficient use of space in #252 where you can see two examples of apps with summarized charts/tables. IMO 18 °C is more self-speaking to me than "yellow but not as yellow than 24 °C". Having both the text and the color is better. Anyway, it's not open for pull requests, sorry. We will design our own way, if we ever do it. Please really consider making your own repo rather than modifying Breezy Weather directly, it's really not much different and it will be easier to maintain.
